pkg/fixtures/named.go

```diff
@@ -0,0 +1,150 @@
+package fixtures
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/justtrackio/gosoline/pkg/funk"
+	"github.com/justtrackio/gosoline/pkg/mdl"
+	"github.com/justtrackio/gosoline/pkg/mdlsub"
+)
+
+type NamedFixture struct {
+	Name  string
+	Value interface{}
+}
+
+type NamedFixtureSet []*NamedFixture
+
+func (l *NamedFixtureSet) All() []interface{} {
+	values := make([]interface{}, 0)
+
+	for _, named := range *l {
+		values = append(values, named.Value)
+	}
+
+	return values
+}
+
+func (l *NamedFixtureSet) Len() int {
+	return len(*l)
+}
+
+func (l *NamedFixtureSet) CountIf(f func(elem interface{}) bool) int {
+	count := 0
+
+	for _, elem := range *l {
+		if f(elem.Value) {
+			count++
+		}
+	}
+
+	return count
+}
+
+func (l *NamedFixtureSet) FindFirst(f func(elem interface{}) bool) (interface{}, bool) {
+	for _, elem := range *l {
+		if f(elem.Value) {
+			return elem.Value, true
+		}
+	}
+
+	return nil, false
+}
+
+func (l *NamedFixtureSet) FindAll(f func(elem interface{}) bool) []interface{} {
+	a := make([]interface{}, 0)
+	for _, elem := range *l {
+		if f(elem.Value) {
+			a = append(a, elem.Value)
+		}
+	}
+
+	return a
+}
+
+func (l *NamedFixtureSet) GetValueByName(name string) interface{} {
+	fixture, ok := funk.FindFirstFunc(*l, func(item *NamedFixture) bool {
+		return item.Name == name
+	})
+	if !ok {
+		panic(fmt.Errorf("failed to get value by name"))
+	}
+
+	return fixture.Value
+}
+
+func (l *NamedFixtureSet) GetValueById(id interface{}) interface{} {
+	fixture, ok := funk.FindFirstFunc(*l, func(item *NamedFixture) bool {
+		valueId, ok := GetValueId(item.Value)
+
+		return ok && id == valueId
+	})
+
+	if !ok {
+		panic(fmt.Errorf("failed to get value by id"))
+	}
+
+	return fixture.Value
+}
+
+func GetValueId(value interface{}) (interface{}, bool) {
+	if kvValue, ok := value.(KvStoreFixture); ok {
+		return GetValueId(kvValue.Value)
+	}
+
+	if kvValue, ok := value.(*KvStoreFixture); ok && kvValue != nil {
+		return GetValueId(kvValue.Value)
+	}
+
+	if identifiable, ok := value.(mdl.Identifiable); ok {
+		return mdl.EmptyIfNil(identifiable.GetId()), true
+	}
+
+	if model, ok := value.(mdlsub.Model); ok {
+		switch modelId := model.GetId().(type) {
+		case string:
+			return modelId, true
+		case *string:
+			return mdl.EmptyIfNil(modelId), true
```

Comment:
could you not just check if its a pointer in general here (maybe via reflection) ? if yes return the value with an `EmptyIfNil()` before otherwise without?

or is that to slow?

Reply:
mdl.EmptyIfNil doesn't work with reflected values. I added some more generic code, please have another look